### ISSUE TITLE
Integrate BKD-based geo shape indexing

### ISSUE THREAD

**Issue by proddata** [crate/crate#11993]
**Use case**:
Significantly improve the indexing and storage performance for geoshapes in CrateDB by using a simple to configure BKD-tree based indexing instead of complex to configure prefix tree based index 

**Feature description**:
Implement Apache Lucenes LatLonShape as default indexing approach for Geoshapes in CrateDB

---

**References**:

- https://lucene.apache.org/core/9_4_0/core/org/apache/lucene/document/LatLonShape.html

Bkd-Tree: A Dynamic Scalable kd-Tree
https://users.cs.duke.edu/~pankaj/publications/papers/bkd-sstd.pdf

Elasticsearch GitHub - BKD-based geo shapes summary:
https://github.com/elastic/elasticsearch/issues/32039

Geospatial Indexing & Search at Scale with Lucene - OGC Portal
https://portal.ogc.org/files/?artifact_id=90337

Deferred patches which might be relevant:

```
- [d] 7e627d27e5c Geo: move indexShape to AbstractGeometryFieldMapper.Indexer (#44979)
- [d] e3adedf610d Geo: implement proper handling of out of bounds geo points (#47734)
- [d] f9cb29450ec Geo: Fixes indexing of linestrings that go around the globe (#47471)
- [d] 2e7d62c27c9 Geo: improve handling of out of bounds points in linestrings (#47939)
- [d] 4ab71116688 Geo: fix indexing of west to east linestrings crossing the antimeridian (#46601)
- [d] 13a8835e5a8 Geo: Change order of parameter in Geometries to lon, lat (#45332)
- [d] f603f06250a Geo: refactor geo mapper and query builder (#44884)
- [d] 33ad7928fbb Geo: extract dateline handling logic from ShapeBuilders (#44187)
- [d] 94b684630c8 [GEO] Refactor DeprecatedParameters in AbstractGeometryFieldMapper (#44923)
- [d] b07310022d2 [SPATIAL] New ShapeFieldMapper for indexing cartesian geometries (#44980)
- [d] 8585d58b767 Provide better error when updating geo_shape field mapper settings (#47281)
```

---

**Comment by mkleen** [comment]
This is a good idea as it would result in smaller index storage and faster queries.

**Comment by BaurzhanSakhariev** [comment]
Implemented in https://github.com/crate/crate/pull/14942 and will be available CrateDB 5.6.

Thanks @piotrrzysko for the contribution!
